specification/consumption/resource-manager/Microsoft.Consumption/preview/2019-04-01-preview/consumption.json

```diff
@@ -0,0 +1,3146 @@
+            "$ref": "./examples/UsageDetailsListForBillingPeriodByEnrollmentAccount.json"
+          },
+          "ManagementGroupUsageDetailsList": {
+            "$ref": "./examples/UsageDetailsListByManagementGroup.json"
+          },
+          "ManagementGroupUsageDetailsListForBillingPeriod": {
+            "$ref": "./examples/UsageDetailsListForBillingPeriodByManagementGroup.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/scopeUsageDetailsParameter"
+          },
+          {
+            "name": "$expand",
+            "description": "May be used to expand the properties/additionalInfo or properties/meterDetails within a list of usage details. By default, these fields are not included when listing usage details.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter usageDetails by properties/resourceGroup, properties/resourceName, properties/resourceId, properties/chargeType, properties/reservationId or tags. The filter supports 'eq', 'lt', 'gt', 'le', 'ge', and 'and'. It does not currently support 'ne', 'or', or 'not'. Tag filter is a key value pair string where key and value is separated by a colon (:).",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$skiptoken",
+            "description": "Skiptoken is only used if a previous operation returned a partial result. If a previous response contains a nextLink element, the value of the nextLink element will include a skiptoken parameter that specifies a starting point to use for subsequent calls.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$top",
+            "description": "May be used to limit the number of results to the most recent N usageDetails.",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "minimum": 1,
+            "maximum": 1000
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/metricParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/UsageDetailsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Consumption/usageDetails/download": {
```

Comment:
This URL references an instance of usageDetails named download. This cannot be used the way you're intending. I think what you're trying to do here is download the usageDetail collection but there is no pattern in ARM for /actions on collections, only /actions on instances.
 
We can either include the download url in the response to usageDetails or add POST /{scope}/providers/Microsoft.Consumption/downloadUsageDetails

Reply:
Fwiw, I had the same thought when I first saw this. That said, this already exists today on older API versions. Instead of changing this only to change it again in a few months, let's please keep this one as-is and fix this when we add download support to Microsoft.CostManagement/query.

Reply:
I don't know if following bad patterns is a great excuse. This will flag your RP as being non compliant with the RPC.

https://armwiki.azurewebsites.net/rp_onboarding/RPCstrongvalidation.html

Which API are you referring too?

Reply:
@RyanBensonMSFT I am not following this feedback . What exactly is getting non-compliant with RPC ?
I see your point that at present in ARM, looks like there is no pattern with actions on collections.
Does this mean, since a pattern does not exist, we should not introduce it all. May be there was never a need for it and now we have a scenario for such actions.
The standard pattern for OData for adding action on collections is https://docs.microsoft.com/en-us/aspnet/web-api/overview/odata-support-in-aspnet-web-api/odata-v3/odata-actions#binding-an-action-to-an-entity-set
This makes sense for our users and it's a lot simpler to understand APIs, when we stick to standards. 
Please let me know, if I am missing something.